### DISCUSSION TITLE
Possible to setup with exisiting glfw instance?

### DISCUSSION THREAD

**Discussion by bwiggs** [AllenDang/cimgui-go#226] [Q&A]
I'm using `github.com/go-gl/glfw/v3.2/glfw` on a current project. I'd like to integrate cimgui-go into it, but im not sure how to provide my existing glfw instance to cimgui-go, or if that's even possible. Anybody have any tips?

**Comment by kkevinchou** [comment]
I'm in a similar situation building a small game. Ideally cimgui-go could expose a more modular drawing method that's responsible for just rendering all of the imgui related elements with the current context. We would retain control of the rendering loop and would then call the cimgui-go related drawing methods. This allows cimgui-go to be more modular and makes it easier for us to adopt other libraries, switch, etc.

Perhaps there is already a path to do this that we're not aware of? @AllenDang @gucio321

**Reply by gucio321** [reply]
generally, to do this, you need to implement `Backend` interface but i don't think all required stuff is already present in go version (especially I mean ImGui_impl_XXXX functions).
But if it gets exported to go it will be absolutely possible.

Pull requests are welcome ;-)
